broker-core/src/main/java/org/apache/qpid/server/security/group/FileGroupDatabase.java

```diff
@@ -104,17 +105,21 @@ public synchronized void setGroupFile(String groupFile) throws IOException
     @Override
     public synchronized void addUserToGroup(String user, String group)
     {
-        Set<String> users = _groupToUserMap.get(keySearch(_groupToUserMap.keySet(), group));
-        if (users == null)
+        Set<String> groupUsers = _groupToUserMap.get(keySearch(_groupToUserMap.keySet(), group));
+        if (groupUsers == null)
         {
             throw new IllegalArgumentException("Group "
                                                + group
                                                + " does not exist so could not add "
                                                + user
                                                + " to it");
         }
+        else if (groupUsers.contains(keySearch(_userToGroupMap.keySet(), user)))
+        {
+            throw new IllegalConfigurationException(String.format("Group member with name'%s' already exists", user));
+        }
 
-        users.add(keySearch(users, user));
+        groupUsers.add(keySearch(groupUsers, user));
 
         Set<String> groups = _userToGroupMap.get(keySearch(_userToGroupMap.keySet(), user));
```

Comment:
let's swap a call to `keySearch(_userToGroupMap.keySet(), user)` with a local variable (as per comments above)

Reply:
Done!